src/print.rs

```diff
@@ -413,10 +414,19 @@ fn all_modules_uniq(module_list: &BTreeSet<String>) -> Vec<String> {
 /// and the list of all modules used in a format string
 fn load_formatter_and_modules<'a>(context: &'a Context) -> (StringFormatter<'a>, BTreeSet<String>) {
     let config = &context.root_config;
-
     let lformatter = StringFormatter::new(&config.format);
     let rformatter = StringFormatter::new(&config.right_format);
     let cformatter = StringFormatter::new(&config.continuation_prompt);
+    let (pformatter, profile_name) = match &context.target {
+        Target::Profile(name) => (
+            match config.profiles.get(name) {
+                Some(format) => StringFormatter::new(format),
+                _ => Err(StringFormatterError::Custom("Invalid Profile".to_string())),
+            },
+            Some(name),
+        ),
+        _ => (StringFormatter::new(""), None),
+    };
```

Comment:
If this isn't too difficult to refactor, I would prefer If only the required targets get initialized like in the previous version of your code.

Reply:
sure that make total sense!